Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -223,17 +223,18 @@ public class HTTPClient {
             )
         }
         let errorStorage: NIOLockedValueBox<Error?> = NIOLockedValueBox(nil)
-        let continuation = DispatchWorkItem {}
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
```

Comment:
Nit: this should probably use the condition variable workaround we used in NIO to avoid triggering the main thread checker.

Reply:
Ah yeah, good idea